_posts/2020-04-01-Robust Apps for the cloud.md

```diff
@@ -0,0 +1,253 @@
+
+As mentioned above, your instances are expected to and will restart. They will be upgraded and will sometimes suffer from file server movements. However you can make your app resilient to all these incidents. In order to guarantee the maximum uptime for your app, **please ensure that you follow all practices**.
+
+## Use Multiple Instances
+
+Running your app on only one VM instance is an immediate single point-of-failure. By ensuring that you have multiple instances allocated to your app, if something goes wrong with any particular instance, your app will still be able to respond to requests going to the other instances. Keep in mind that your **app code should be able to handle multiple instances** without synchronization issues when reading from or writing to data sources. You can allocate multiple instances to your app using the "Scale out (App Service Plan)" blade:
+
+![multiple-instances]({{site.baseurl}}/media/2020/04/multiple-instances.png)
+
+To avoid a single point-of-failure, run your app with **at least 2-3 instances**. This is especially important if your app takes considerable time to start (known as cold start). Running more than one instance ensures that your application is available when App Service moves or upgrades the underlying VM instances. You can also configure rules to automatically scale out based on predefined rules such as:
+
+- Time of day (when the app has the most traffic)
+- Resource utilization (memory, CPU, etc.)
+- A combination of both!
+
+**Learn More**
+
+- [Get started with Autoscale in Azure](https://docs.microsoft.com/azure/azure-monitor/platform/autoscale-get-started?toc=/azure/app-service/toc.json)
+- [App Service Warm-Up Demystified](https://michaelcandido.com/app-service-warm-up-demystified/)
+
+## Update your default settings
+
+App Service has many settings for developers to configure the web app to their use case. **Always-On** keeps your VM instances alive even when no requests have been received in the last 20 minutes. By default, Always-On is disabled; enabling Always-On will limit application cold starts. **ARR Affinity** creates sticky sessions so that clients will connect to the same app instance on subsequent requests. However, ARR Affinity can cause unequal distribution of requests between your instances and possibly overload an instance. For production apps that are aiming to be robust, it is recommended to set **Always on to On** and **ARR Affinity to Off**. Disabling ARR Affinity assumes that your application is either stateless, or the session state is stored on a remote service such as a cache or database.
```

Comment:
I left a note here about disabling ARR Affinity. If an app uses cookie-based auth, then disabling ARR Affinity requires them to move that state to a remote store.

Reply:
Agreed, thanks!